protx/decorators.py

```diff
@@ -3,27 +3,48 @@
 from diskcache import Cache
 import os
 import logging
-from flask import request
+from flask import request, redirect
 import requests
 
 logger = logging.getLogger(__name__)
 
 cache = Cache("database_cache")
 
 
+def is_setup_complete() -> bool:
+    # making request directly to core django. alternatively, we could
+    # make the request to the request.get_host() (but then a bit
+    # tricky if it is local development and accessing cep.dev)
+    r = requests.get("http://django:6000/api/workbench/", cookies=request.cookies)
+    r.raise_for_status()
+    json_response = r.json()
+    return json_response['response']['setupComplete']
+
+
+def onboarded_user_setup_complete(function):
+    """Decorator requires user to have setup_completed or redirects.
+    """
+    @wraps(function)
+    def wrapper(*args, **kwargs):
+        try:
+            if not is_setup_complete():
+                redirect_url = request.url_root.split(',')[0] if ',' in request.url_root else request.url_root
+                return redirect(redirect_url + '/workbench/onboarding')
+        except Exception as e:
+            logger.error(e)
+            raise Forbidden
```

Comment:
suggestion:  right now we have the pages in cms and usually configured that login is required there.  So its hard for me to imagine us hitting this code.  but if that ever changes (i.e. we don't have those pages blocked by the cms), should already polish this so that it redirects to /login here instead of forbidden?  (ideally with `next` but we assume that isn't working at the moment with cms/core)

not certain but if we do that, let's update docstr and drops the logger.error?

or if we ignore, can we add a comment here  like ```# User isn't logged in which is unexpected as pages are controlled by CMS)```

Reply:
Pretty sure the `/accounts/login` redirect is the default behavior of the CMS, but we overrode that in the portal to force users to be logged into the CMS if they log into the portal (portal auths against the CMS as the user that just authed). That way page behavior is still controlled by the CMS. SO I don't see any circumstances under which we would ever hit this code (either pages are public or they require login, which the portal forces on the cms in the current v2 architecture). The Forbidden is used by the CMS I believe. We should investigate before making any changes because _thar be dragons here_.

Reply:
> this shouldn't happen anymore with this PR, can you walk me through this. so you were an unauthed user going directly to which route?

Ignore this - was on the wrong branch then.